### PR TITLE
[ANDROID] Order Details

### LINKED ISSUE #387: [ANDROID] Order Item Extension
> Description

In Order Page, order items are not clickable. They have to navigate user to the all items ordered in that order action. A recycler view can be implemented for showing all items in that list.

**_Deadline_: 24.01.2021, 23:59**

### PR DESCRIPTION
**Pull request by handesirikci**
## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Description

Before this feature, user was able to just see her orders as a list where all items in an order item can not seem separately. And for delivered orders "review" option was not enabled.

By this feature,

- User can see one order in an order item and click on it
- When she clicks on an order, all items in that order can be seem in an other fragment
- User can see amounts of the products and total price
- User can see review option for delivered orders

# Screenshots

https://user-images.githubusercontent.com/49978504/105627054-54990180-5e45-11eb-86f1-9302d342950b.mp4

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have checked my code and corrected any misspellings
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by asliayk — CHANGES_REQUESTED** [review]
I think there are some missing parts:
- There is no date info for order. It will be better if the user sees the order date.
- There's also no image for product (You have used mock data). You can use Picasso library, as we have used in CartFragment to show the image of the product.
- There's also no amount info for each product. I think the user should see the amount info.

It will be better if you also add these things.

File: ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/adapters/OrderProductsAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    private val orderPurchasedList: ArrayList<OrderPurchasedModel>?
+) : ListAdapter<OrderPurchasedModel, OrderProductsAdapter.RowHolder>(OrderProductsDiffCallback()) {
+
+
+
+    class RowHolder(val binding: OrderProductItemBinding) : RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(orderPurchased: OrderPurchasedModel, position: Int) {
+            binding.listproduct = orderPurchased
+        }
```
Comment (karalleyna) [inline ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/adapters/OrderProductsAdapter.kt]: After you initialize the data in the layout, you don't the bind the price of the order neither in the bind function of RowHolder nor in the `order_product_item.xml`

File: ecommerce/android/Getflix/app/src/main/res/layout/order_product_item.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            <TextView
+                android:id="@+id/total_price_cart_product"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginEnd="16dp"
+                android:layout_marginRight="15dp"
+                android:layout_marginBottom="15dp"
+                android:fontFamily="@font/open_sans_condensed_bold"
+                android:gravity="right"
+                android:text="50$"
```
Comment (karalleyna) [inline ecommerce/android/Getflix/app/src/main/res/layout/order_product_item.xml]: Actually, our currency is **"TL"** not **"$"**

File: ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/fragments/OrderProductsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        var product= ProductModel(27, "Samsung S20 Ultra", 10999.0, "2020-12-26T10:47:38.961041Z", 50, 11, 50,
+            "Ekran Boyutu: 6.2', Ekran Çözünürlüğü: 1440x3200 px, Arka Kamera: 12 MP, Üçlü Kamera, Ön Kamera: 10 MP, 4G, Dahili Hafıza: 128 GB",
+            SubcategoryModel("Cell Phones & Accessories", 2),
+            "Galaxy S serisi akıllı cep telefonlarıyla nefes kesici teknolojik yenilikleri sergileyen Samsung, sinematik kare/saniye oranlarında ve 8K çözünürlükte video kaydı yapan",
+            0.045 , CategoryModel("Electronics", 1, list2), BrandModel("Samsung",18), VendorModel(0.0,3, "Can Batuk"),
+            4.545454545454546,
+            list1, 10504.045, false)
+        var purchasedModel = OrderPurchasedModel(1, 1, product, "",1,"",VendorModel(0.0,3, "Can Batuk"),address)
+        val listproducts = arrayListOf(purchasedModel)
+        val listAdapter = OrderProductsAdapter(listproducts)
```
Comment (karalleyna) [inline ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/fragments/OrderProductsFragment.kt]: It should be improved to get the orders from the respond of the corresponding function in our `GetflixApiService.kt` not the mock data.

File: ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/adapters/OrderProductsAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            holder.bind(it, position)
+
+        }
+    }
+
+}
+
+class OrderProductsDiffCallback : DiffUtil.ItemCallback<OrderPurchasedModel>() {
+    override fun areItemsTheSame(oldItem: OrderPurchasedModel, newItem: OrderPurchasedModel): Boolean {
+        return oldItem.id == newItem.id
```
Comment (karalleyna) [inline ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/adapters/OrderProductsAdapter.kt:64]: Actually, I like the idea of the comparison which looks the ids of the orders in spite of 
all their properties one by one to determine whether two `OrderPurchasedModel` instances are same or not .

**Review by karalleyna — CHANGES_REQUESTED** [review]
I believe our design should be focused on the expectation of our customers. However, customer cannot know how many products she or he bought in one of his or her order if we don't change it now. Also, I did not review your previous PR in regard to the orders of customer. So, I noticed you did not set the scene of the status of the order. For example, `at_cargo` is shown to the user as its coming from the backend.  Also, could you explain the functionality of the first page to me? I did not understand the purposes of pages since if the first page included the overall look of the products they have bought in one shot, it would not have a title same to the name of the product.

Other than these issues, it seems correct. So, we can merge after you changed them.

File: ecommerce/android/Getflix/app/src/main/res/layout/order_product_item.xml (hunk 2 above)
Comment (handesirikci) [reply ecommerce/android/Getflix/app/src/main/res/layout/order_product_item.xml]: it was just a mock data not the actual one

File: ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/adapters/OrderProductsAdapter.kt (hunk 1 above)
Comment (handesirikci) [reply ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/adapters/OrderProductsAdapter.kt]: now i am pushing the revised version which binded

File: ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/fragments/OrderProductsFragment.kt (hunk 3 above)
Comment (handesirikci) [reply ecommerce/android/Getflix/app/src/main/java/com/example/getflix/ui/fragments/OrderProductsFragment.kt]: it is on the to do list

**Comment by handesirikci** [comment]
Thanks for all of you. I pushed the new version and i think it is OK now

**Pull request merged**